File: julia/bubblesort.jl
"""
BUBBLE SORT!
"""
function bubblesort(lst)
  n = length(lst)
  while true
    swapped = false
    for i = 2:(n)
      if lst[i-1] > lst[i]
        lst[i-1], lst[i] = lst[i], lst[i-1]
        swapped = true
      end
    end

    if !swapped
      break
    end
  end
end

File: javascript/bubblesort.js
function bubbleSort(lst) {
  n = lst.length;
  while (true) {
    let swapped = false
    for (let i = 1; < n+1; i++) {
      let tmp = lst[i-1]
      lst[i-1] = lst[i]
      lst[i] = tmp
      swapped = true
    }

    if (!swapped) {
      break
    }
  }
}
